### PR TITLE
difftest: Provides a script to change the gatewayendpoint to interface

### PR DESCRIPTION
**Pull request by xiaokamikami**
Provides a script to modify the gatewayendpoint to the interface connection way, for the difftest connection process of the ST environment, and supports the removal of simv top and the removal of simv top from filelist

### CODE REVIEW

**Comment by poemonsense** [comment]
CI fails because of Spike changes. I'll see what's wrong with Spike.

**Comment by poemonsense** [comment]
I'll re-run this after merging https://github.com/OpenXiangShan/riscv-isa-sim/pull/76

File: scripts/st_tools/interface.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
+#/usr/bin/python3
+# -*- coding: UTF-8 -*-
+
+#***************************************************************************************
+# Copyright (c) 2024 Beijing Institute of Open Source Chip (BOSC)
+#
+# XiangShan is licensed under Mulan PSL v2.
```
Comment (poemonsense) [inline scripts/st_tools/interface.py:7]: DiffTest is licensed ...

**Review by poemonsense — COMMENTED** [review]
Can you describe how we can use this file in the command line and what it does in this PR? Do we need to add a test for it?

File: scripts/st_tools/interface.py (hunk 1 above)
Comment (xiaokamikami) [reply scripts/st_tools/interface.py]: ok

**Comment by xiaokamikami** [comment]
> Can you describe how we can use this file in the command line and what it does in this PR? Do we need to add a test for it?

I will add the corresponding CI together with adding CI to Xiangshan repository

**Review by poemonsense — APPROVED** [review]

**Pull request merged**